core/dbt/clients/jinja.py

```diff
@@ -354,5 +354,7 @@ def undefined_error(msg):
     raise jinja2.exceptions.UndefinedError(msg)
 
 
-def extract_toplevel_blocks(data):
-    return BlockIterator(data).lex_for_blocks()
+def extract_toplevel_blocks(data, allowed=None, collect_raw_data=True):
+    bi = BlockIterator(data)
+    return bi.lex_for_blocks(allowed=allowed,
+                             collect_raw_data=collect_raw_data)
```

Comment:
can you either collapse this back or name `bi` something else? so minor but don't use acronyms/single letter variables

Reply:
I'm having trouble understanding the utility of `collect_raw_data`. i mean, I understand what it does, but I'm not sure when we'd use it in practice. all the calls to `extract_toplevel_blocks` in this PR use collect_raw_data=False

Reply:
The key is in the things I didn't change in this PR! We use it here during RPC parsing: https://github.com/fishtown-analytics/dbt/blob/fix/block-parser-ignore-more/core/dbt/task/compile.py#L64

Reply:
@beckjake thanks, that makes sense. and the more code i read here the more it makes sense why we'd want it. it's nice to be able to turn this on and off for RPC use cases, tests, and also potentially for the evolution of the dbt parser. it gives us lots of options.

can you rename `allowed` to `allowed_blocks` and add a docstring to this fn?